checkov/sca_package/scanner.py

```diff
@@ -2,103 +2,110 @@
             # PYCHARM_HOSTED env variable equals 1 when running via Pycharm.
             # it avoids us from crashing, which happens when using multiprocessing via Pycharm's debug-mode
             logging.warning("Running the scans in sequence for avoiding crashing when running via Pycharm")
             scan_results = []
-            for curr_arg in args:
-                scan_results.append(await self.run_scan(*curr_arg))
+            for input_path in input_paths:
+                scan_results.append(self.run_scan(input_path))
         else:
-            async with Pool() as pool:
-                scan_results = await pool.starmap(self.run_scan, args)
+            input_paths = [(input_path,) for input_path in input_paths]
+            with Pool() as pool:
+                scan_results = pool.starmap(self.run_scan, input_paths)
 
         return scan_results
 
-    async def run_scan(
-        self,
-        command: str,
-        input_path: Path,
-        output_path: Path,
-    ) -> Dict[str, Any]:
+    def run_scan(self, input_path: Path) -> dict:
         logging.info(f"Start to scan package file {input_path}")
-        process = await asyncio.create_subprocess_shell(
-            command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+
+        headers = merge_dicts(
+            get_default_get_headers(bc_integration.bc_source, bc_integration.bc_source_version),
+            {"Authorization": bc_integration.get_auth_token()},
         )
 
-        stdout, stderr = await process.communicate()
+        request_body = {
+            "compressedFileBody": compress_file_gzip_base64(str(input_path)),
+            "compressionMethod": "gzip",
+            "fileName": input_path.name
+        }
 
-        # log output for debugging
-        logging.debug(stdout.decode())
+        response = requests.request(
+            "POST", f"{self.base_url}/api/v1/vulnerabilities/scan", headers=headers,
+            data=request_body
+        )
 
-        exit_code = await process.wait()
+        response.raise_for_status()
+        response_json = response.json()
 
-        if exit_code:
-            logging.error(f"Failed to scan package file {input_path}")
-            logging.error(stderr.decode())
-            return {}
+        if response_json["status"] == "already_exist":
+            return json.loads(
+                decompress_file_gzip_base64(
+                    response_json["outputData"]
+                )
+            )
+
+        return self.run_scan_busy_wait(response_json['id'])
+
+    def run_scan_busy_wait(self, scan_id: str) -> dict:
+        current_state = "Empty"
+        desired_state = "Result"
+
+        headers = merge_dicts(
+            get_default_get_headers(bc_integration.bc_source, bc_integration.bc_source_version),
+            {"Authorization": bc_integration.get_auth_token()},
+        )
+        response = requests.Response()
+
+        while current_state != desired_state:
```

Comment:
add some limitation. (e.g: after 2 minutes, abort and log it)

Reply:
Agree with @itai1357 , Let's limit it to 60 seconds please

Reply:
@Eliran-Turgeman just making sure you fixed it

Reply:
Yeah done, refer to lines 95 - 100